pgmpy/factors/Factor.py

```diff
@@ -191,39 +190,22 @@ def marginalize(self, variables, inplace=True):
             factor = self
         else:
             factor = Factor(self.scope(), self.cardinality, self.values)
-
+        marginalize_index = np.array(np.where(np.in1d(factor.scope(), variables)))
+        assign = np.array(factor.cardinality)
+        assign[marginalize_index] = -1
+        marg_factor = []
```

Comment:
@palashahuja I think we can have a better name for `marg_factor`. Maybe `new_values`? Any better name in your mind?